src/app/pages/token/token-detail/token-content/token-content-tab/token-holders-tab/token-holders-tab.component.ts

```diff
@@ -187,4 +204,45 @@ export class TokenHoldersTabComponent implements OnInit {
       this.getHolderNFT();
     } catch (error) {}
   }
+
+  getDenomHolder() {
+    this.tokenService.getDenomHolder(this.tokenDetail?.denomHash, this.keyWord || null).subscribe({
+      next: (res) => {
+        this.totalHolder = this.keyWord?.length > 0 ? res?.account?.length : this.tokenDetail?.holder;
+        if (this.totalHolder > this.numberTopHolder) {
+          this.pageData.length = this.numberTopHolder;
+        } else {
+          this.pageData.length = this.totalHolder;
+        }
+
+        let dataFlat = res.account?.map((item) => {
+          let amount = item.balances?.find((k) => k.denom === this.tokenDetail?.denomHash)?.amount;
+
+          return {
+            owner: item.address,
+            amount,
+            balance: amount,
+            percent_hold: new BigNumber(amount).dividedBy(this.tokenDetail?.totalSupply).multipliedBy(100),
+            value:
+            new BigNumber(amount)
+                .multipliedBy(this.tokenDetail?.price)
+                .dividedBy(Math.pow(10, this.decimalValue))
```

Comment:
`Math.pow(10, this.decimalValue)` chuyển sang BigNumber luôn:
`BigNumber(10).pow(...)`